File: backend/Model/TaskItem.cs
using System.Text.Json.Serialization;

namespace Model
{
    public enum TaskItemStatus
    {
        Todo,
        Doing,
        Done
    }

    public class TaskItem
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public DateTime DueDate { get; set; }

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public TaskItemStatus Status { get; set; } = TaskItemStatus.Doing;
    }
}

File: frontend/src/models/TaskItem.ts
import moment from "moment"
moment.locale("es")

export type TaskStatus = "Todo" | "Doing" | "Done"

export class TaskItem {
    id!: number
    title!: string
    description!: string
    status!: TaskStatus
    dueDate!: Date
    dueDateString!: string
    isDueDate:boolean

    constructor(id:number, tile: string, description: string, dueDate: Date, status: TaskStatus = "Todo" ) {
        this.id = id;
        this.title = tile;
        this.description = description;
        this.status = status;
        this.dueDate = dueDate;
        this.dueDateString = moment(dueDate).format("DD-MMM-YYYY")
        this.isDueDate = moment(dueDate).isBefore(moment(), "day")
    }
    
}